lib/logstash/inputs/jms.rb

```diff
@@ -109,34 +134,125 @@ class LogStash::Inputs::Jms < LogStash::Inputs::Threadable
   public
   def register
     require "jms"
-    @connection = nil
 
-    if @yaml_file
-      @jms_config = YAML.load_file(@yaml_file)[@yaml_section]
+    check_config
+    load_ssl_properties
+    load_system_properties if @system_properties
+    @jms_config = jms_config
+
+    @logger.debug("JMS Config being used ", :context => obfuscate_jms_config(@jms_config))
+  end # def register
+
+  def obfuscate_jms_config(config)
+    config.each_with_object({}) { |(k, v), h| h[k] = obfuscatable?(k) ? 'xxxxx' : v }
+  end
+
+  def obfuscatable?(setting)
+    [:password, :keystore_password, :truststore_password].include?(setting)
+  end
+
+  def jms_config
+    return jms_config_from_yaml(@yaml_file, @yaml_section) if @yaml_file
+    return jms_config_from_jndi if @jndi_name
+    jms_config_from_configuration
+  end
 
-    elsif @jndi_name
-      @jms_config = {
-        :require_jars => @require_jars,
-        :jndi_name => @jndi_name,
-        :jndi_context => @jndi_context}
 
-    elsif @factory
-      @jms_config = {
+  def jms_config_from_configuration
+    config = {
         :require_jars => @require_jars,
         :factory => @factory,
         :username => @username,
-        :password => @password,
         :broker_url => @broker_url,
         :url => @broker_url # "broker_url" is named "url" with Oracle AQ
-      }
+    }
+    config.merge!(:password => @password.value) unless @password.nil?
```

Comment:
I find this easier to read and it removes the need for `merge!` ...
```
    config[:password] = @password.value unless @password.nil?
    correct_factory_hash(config, @factory_settings) unless @factory_settings.nil?
    config
  end

  def correct_factory_hash(original, value)
    if hash.is_a?(String)
      return true if value.downcase == "true"
      return false if value.downcase == "false"
    end
    if value.is_a?(Hash)
      value.each { |key, value| original[key.to_sym] = correct_factory_hash({}, value) }
      return original
    end
    value
  end
```

Reply:
Agreed, and done